ASP/ASP.sublime-syntax

```diff
@@ -2,123 +2,662 @@
+    - include: comments
+
+  not_end_of_statement:
+    - match: ':'
+      scope: invalid.illegal.unexpected-end-of-statement.asp
+      pop: true
+    - match: \n
+      scope: invalid.illegal.unexpected-end-of-statement.asp
+      pop: true
+    - match: '{{rem_comment_begin}}|{{apostrophe_comment_begin}}'
+      set:
+        - meta_scope: invalid.illegal.unexpected-end-of-statement.asp
+        - match: \n
+          pop: true
+    - include: allow_line_continuation
+
+  expect_not_end_of_statement:
+    - match: '' # this match pattern has been added to disable the illegal highlighting where a statement ends unexpectedly - to more closely match other syntaxes
+      pop: true
+    - include: not_end_of_statement
+    - match: '\s*(?=\S)'
+      pop: true
+
+  expect_identifier_reference:
+    - match: '\s+'
+      scope: ''
+    - match: '{{identifier}}'
+      captures:
+        0: variable.other.asp
+      #  1: invalid.illegal.unclosed-variable-identifier.asp
+      pop: true
+    - match: '' # this match pattern has been added to disable the illegal highlighting where a statement ends unexpectedly - to more closely match other syntaxes
+      pop: true
+    - include: unexpected_token
+
+  inside_string:
+    - meta_scope: string.quoted.double.asp
+    - include: root_asp
+    - match: \n
+      scope: invalid.illegal.unclosed-string.asp
+      pop: true
+    - match: '""'
+      scope: constant.character.escape.apostrophe.asp
+    - match: '"'
+      scope: punctuation.definition.string.end.asp
+      pop: true
+
+  illegal_names:
+    - match: '{{keywords}}|{{constants}}|{{reserved_words}}|{{functions}}|{{asp_builtin_classes}}'
+      scope: invalid.illegal.name.asp
+      pop: true
+
+  allow_statement_separator:
+    - match: ':'
+      scope: punctuation.terminator.statement.asp
+      pop: true
+
+  unexpected_token:
+    - match: \n
+      scope: invalid.illegal.missing-token.asp
+      pop: true
+    - match: '\S+'
+      scope: invalid.illegal.unexpected-token.asp
+      pop: true
+
+  statements:
+    - include: variable_definitions
+    - include: allow_line_continuation
+    - match: '\b(?i:On\s+Error\s+)'
       scope: storage.type.asp
-    - match: (?i:\b(Private|Public|Default)\b)
+      push:
+        - include: not_end_of_statement
+        - match: '\b(?i:Resume\s+Next){{whitespace_or_end_of_statement}}'
+          scope: storage.type.asp
+          pop: true
+        - match: '\b(?i:Goto\s+0){{whitespace_or_end_of_statement}}'
+          scope: storage.type.asp
+          pop: true
+        - include: unexpected_token
+    - match: '\b(?i:Randomize\s+Timer){{whitespace_or_end_of_statement}}'
+      scope: storage.type.asp
+    - match: '\b(?i:Call|Set){{whitespace_or_end_of_statement}}'
+      scope: storage.type.asp # maybe this should be a keyword instead?
```

Comment:
here is one scope that I am not sure about. In ASP, to set a variable to an object, as opposed to a number or string, it is necessary to use `set var_name = obj_reference` instead of just `var_name =`...
also, `Call` is completely optional, but if used, must be followed by a method name (i.e. a callable object)
so far I have kept the scope as it was previously, but I believe that a `keyword` scope would be more appropriate.  As far as I know, these are unique to VB and ASP. Thoughts?